src/Interpreters/ExecuteScalarSubqueriesVisitor.cpp

```diff
@@ -78,16 +78,19 @@ static auto getQueryInterpreter(const ASTSubquery & subquery, ExecuteScalarSubqu
     subquery_settings.extremes = false;
     subquery_context->setSettings(subquery_settings);
 
-    /// When execute `INSERT INTO t WITH ... SELECT ...`, it may lead to `Unknown columns`
-    /// exception with this settings enabled(https://github.com/ClickHouse/ClickHouse/issues/52494).
-    subquery_context->getQueryContext()->setSetting("use_structure_from_insertion_table_in_table_functions", false);
-    if (!data.only_analyze && subquery_context->hasQueryContext())
+    if (subquery_context->hasQueryContext())
```

Comment:
Is it possible to have non-empty `data.scalars` but no query context? If yes, why do we execute such query on a startup without context?

Reply:
The main problem was in line with `subquery_context->getQueryContext()->setSetting` that used query context without checking if it is even there. Now we add a check for it. Other things did not change

Reply:
Isn't it a problem in itself that we are executing this code without initialized query context?